File: docs/guides/kotlin/src/main/kotlin/com/algolia/snippets/pushSetup.kt
package org.example

import com.algolia.client.api.IngestionClient
import com.algolia.client.configuration.*
import com.algolia.client.model.ingestion.*
import com.algolia.client.transport.*
import java.io.File
import kotlinx.serialization.builtins.ListSerializer
import kotlinx.serialization.json.Json

suspend fun main() {
  val json = File("records.json").readText()
  val records: List<PushTaskRecords> =
    Json.decodeFromString(ListSerializer(PushTaskRecords.serializer()), json)

  // use the region matching your applicationID
  val client =
    IngestionClient(
      appId = "ALGOLIA_APPLICATION_ID",
      apiKey = "ALGOLIA_API_KEY",
      region = "ALGOLIA_APPLICATION_REGION",
    )

  try {
    // setting `watch` to `true` will make the call synchronous
    val resp =
      client.pushTask(
        taskID = "YOUR_TASK_ID",
        pushTaskPayload =
          PushTaskPayload(
            action = Action.entries.first { it.value == "addObject" },
            records = records,
          ),
        watch = true,
      )

    println(resp)
  } catch (e: Exception) {
    println(e.message)
  }
}

File: docs/guides/php/src/pushSetup.php
<?php

require __DIR__.'/../vendor/autoload.php';
use Algolia\AlgoliaSearch\Api\IngestionClient;

$records = json_decode(file_get_contents('records.json') ?: '[]', true);

// use the region matching your applicationID
$client = IngestionClient::create('ALGOLIA_APPLICATION_ID', 'ALGOLIA_API_KEY', 'ALGOLIA_APPLICATION_REGION');

// setting `watch` to `true` will make the call synchronous
$resp = $client->pushTask(
    'YOUR_TASK_ID',
    ['action' => 'addObject',
        'records' => $records,
    ],
    true,
);

echo $resp;
